src/components/FeaturesDemo/CodeEditor/index.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { JSX, ReactElement, ReactNode } from "react";
+
+import SearchIcon from "@site/static/icons/search.svg";
+import CloseIcon from "@site/static/icons/close.svg";
+import AngleIcon from "@site/static/icons/angle.svg";
+import TypescriptIcon from "@site/static/icons/typescript.svg";
+import FolderIcon from "@site/static/icons/folder.svg";
+import TextIcon from "@site/static/icons/text.svg";
+import SassIcon from "@site/static/icons/sass.svg";
+import TestplaneIcon from "@site/static/icons/testplane.svg";
+import TerminalIcon from "@site/static/icons/terminal.svg";
+import { Collapsible } from "@site/src/components/Collapsible";
+import styles from "./styles.module.scss";
+
+export interface TabProps {
+    title: string;
+    icon?: ReactElement;
+    isActive?: boolean;
+}
+
+function Tab(props: TabProps) {
+    return (
+        <div className={clsx(styles["tab-item"], "relative ml-5 flex items-center text-nowrap")}>
+            <div className={clsx({ "text-indigo-200": props.isActive })}>{props.title}</div>
+            <div className={clsx("ml-5 size-3", { "fill-indigo-200": props.isActive })}>
+                {props.icon ?? <CloseIcon />}
+            </div>
+            {props.isActive && <div className="absolute bottom-0 h-0.5 w-full bg-indigo-500"></div>}
+        </div>
+    );
+}
+
+export enum FsTreeItemType {
+    Dir,
+    Ts,
+    Txt,
+    Sass,
+    Testplane,
+}
+
+export interface FsTreeItem {
+    title: string;
+    type?: FsTreeItemType;
+    isCollapsed?: boolean;
+    isModified?: boolean;
+    isSelected?: boolean;
+    children: FsTreeItem[];
+}
+
+function CollapsibleFsTreeItem(props: FsTreeItem) {
+    const getItemIcon = (type?: FsTreeItemType) => {
+        if (type === FsTreeItemType.Dir) {
+            return <FolderIcon className="size-3" />;
+        } else if (type === FsTreeItemType.Ts) {
+            return <TypescriptIcon className="size-3" />;
+        } else if (type === FsTreeItemType.Sass) {
+            return <SassIcon className="size-3" />;
+        } else if (type === FsTreeItemType.Testplane) {
+            return <TestplaneIcon className="size-3" />;
+        } else {
+            return <TextIcon className="size-3" />;
+        }
+    };
+
+    const ItemTitle = (
+        <div
+            className={clsx("ml-5 flex items-center", {
+                "fill-indigo-400 text-indigo-400": props.isModified,
+            })}
+        >
+            <div className="relative z-10">
+                {props.type === FsTreeItemType.Dir && (
+                    <AngleIcon
+                        className={clsx(
+                            "absolute -left-5 top-1/2 flex size-3 -translate-y-1/2 items-center",
+                            { "rotate-90": !props.isCollapsed },
+                        )}
+                    />
+                )}
+            </div>
+            {props.isSelected && (
+                <div className="absolute left-0 z-0 h-6 w-full bg-indigo-900"></div>
```

Comment:
Because I want the selection background width to be 100% of the whole fs tree column. However, each element takes only part of the column's width, so I can't just use background-color and need this artificial element for that.

![](https://i.imgur.com/xGrf385.png)